packages/common/src/storage/helpers/storage-serializer.ts

```diff
@@ -0,0 +1,19 @@
+import BigNumber from 'bignumber.js'
+
+/**
+ * Serializes the given object into a JSON string, with support for BigInt and BigNumber values.
+ * Since BigInt and BigNumber are not supported by JSON.stringify, it serialize them as strings.
+ *
+ * @param toStringify - The object to be serialized.
+ * @returns A stringfied JSON object with BigInt and BigNumber values serialized as strings.
+ */
+export const storageSerializer = (toStringify: unknown) => {
```

Comment:
wrap, JSON.stringify(data, storageSerializer) won't work, it needs to be used directly. storageSerializer(data)